File: civitasRUBY/lib/sorpresa_especulador.rb
#encoding: utf-8

module Civitas
  class SorpresaEspeculador < Sorpresa
    def initialize(civitasJuego)
      super("El jugador pasa a ser Especulador")
      @juego=civitasJuego
    end
    
    #AplicarAJugador de SorpresaEspeculador
    def aplicarAJugador(actual,todos)
      if (jugadorCorrecto(actual, todos) and !todos[actual].esEspeculador)
        informe(actual, todos)
        @juego.jugadorAEspeculadorJug(todos.at(actual))
      end
    end
    
    #toString de SorpresaEspeculador
    def toString
      return "Sorpresa tipo IrACasilla("+self.to_s+")"
    end
    
  end
end

File: civitasJAVA/civitas/src/civitas/SorpresaEspeculador.java
package civitas;

import java.util.ArrayList;


public class SorpresaEspeculador extends Sorpresa {
    CivitasJuego juego;

    //Contstructor    
    SorpresaEspeculador(CivitasJuego _juego)
    {
        this.texto = "Sorpresa convierte a Especulador";
        juego      = _juego;
    }
    
    //Aplica la sorpresa al jugador pasado como parámetro
    void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if (jugadorCorrecto(actual,todos)  &&  !todos.get(actual).esEspeculador()) {
            informe(actual,todos);
            juego.jugadorAEspeculador(todos.get(actual));
        }
    }
}
